extensions/amp-iframely/0.1/amp-iframely.js

```diff
@@ -0,0 +1,349 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {createElementWithAttributes, removeElement} from '../../../src/dom';
+import {isLayoutSizeDefined} from '../../../src/layout';
+import {setStyle, toggle} from '../../../src/style';
+import {userAssert} from '../../../src/log';
+
+/** @const {string} */
+export const TAG = 'amp-iframely';
+const DOMAIN = 'cdn.iframe.ly';
+
+/**
+ * Finds an iFrame that sent a message by its contentWindow.
+ * @param {array} iframes - narrowed set of iFrames to search in
+ * @param {document.window} contentWindow - source of a received message
+ * @return {iframe} the matched iFrame
+ * */
+function findIframeByContentWindow(iframes, contentWindow) {
+  let selectedIframe = false;
+  for (let i = 0; i < iframes.length && !selectedIframe; i++) {
+    const iframe = iframes[i];
+    if (iframe.contentWindow === contentWindow) {
+      selectedIframe = iframe;
+    }
+  }
+  return selectedIframe;
+}
+
+/**
+ * Implementation of the amp-iframely component.
+ *
+ * See {@link ../amp-iframely.md} for the spec.
+ */
+export class AmpIframely extends AMP.BaseElement {
+  /** @param {!AmpElement} element */
+  constructor(element) {
+    super(element);
+
+    this.src_ = null;
+
+    /**
+     * The main Iframely iframe element.
+     * @private {?Element}
+     */
+    this.iframely_ = null;
```

Comment:
nit: just `iframe_`, since it's the only iframe, the main mechanism, and a branded name loses meaning for maintainers unfamiliar.